mlx/backend/metal/metal.cpp

```diff
@@ -74,6 +79,8 @@ std::function<void()> make_task(
       if (arr.is_tracer()) {
         inputs = arr.inputs();
       }
+
+      debug_set_primitive_buffer_label(arr.primitive());
```

Comment:
The goal was to always ensure the correct command buffer is labeled, if one exists.

Reply:
> The goal was to always ensure the correct command buffer is labeled, if one exists.

Yea exactly.. in that case I would pass the buffer explicitly. Accessing it via the primitive is not a guarantee that it's the correct buffer since you are just getting the device's most recently created buffer.. (not necessarily corresponding to the primitive).

In this case it's the same (and I don't expect it to change.. so indeed it's minor)

Reply:
Updated with explicit command buffer argument.